grease-cli/src/Grease/Main.hs

```diff
@@ -940,36 +939,8 @@ simulateMacawSyntax la halloc archCtx simOpts parserHooks = do
   let cfgs' = Map.map (\cfg -> MacawEntrypointCfgs cfg Nothing) cfgs
   let memory = MC.emptyMemory (archCtx ^. archInfo . to MI.archAddrWidth)
   let dl = macawDataLayout archCtx
-  let setupHook :: forall sym. SetupHook sym arch
-      setupHook = Macaw.SetupHook $ \bak mvar funOvs -> do
-        -- Register overrides, both user-defined ones and ones that are
-        -- hard-coded into GREASE itself.
-        forM_ (Map.elems funOvs) $ \mfo -> do
-          let publicOvHdl = Macaw.mfoPublicFnHandle mfo
-              publicOv = Macaw.mfoPublicOverride mfo
-          Stubs.SomeFunctionOverride fnOv <- pure $ Macaw.mfoSomeFunctionOverride mfo
-          C.bindFnHandle publicOvHdl (C.UseOverride publicOv)
-          let auxFns = Stubs.functionAuxiliaryFnBindings fnOv
-          forM_ auxFns $ \(C.FnBinding auxHdl auxSt) -> C.bindFnHandle auxHdl auxSt
-
-        -- In addition to binding function handles for the user overrides,
-        -- we must also redirect function handles resulting from parsing
-        -- forward declarations (`declare`) to actually call the overrides.
-        Macaw.registerMacawSexpProgForwardDeclarations bak la dl mvar funOvs (CSyn.parsedProgForwardDecs prog)
-        forM_ (Map.elems funOvs) $ \mfo ->
-          case Macaw.mfoSomeFunctionOverride mfo of
-            Stubs.SomeFunctionOverride fnOv ->
-              Macaw.registerMacawOvForwardDeclarations bak funOvs (Stubs.functionForwardDeclarations fnOv)
-        forM_ (Map.elems cfgs) $ \entrypointCfgs ->
-          forM_ (startupOvForwardDecs <$> entrypointStartupOv entrypointCfgs) $ \startupOvFwdDecs ->
-            Macaw.registerMacawOvForwardDeclarations bak funOvs startupOvFwdDecs
-
-        -- Register defined functions.
-        forM_ (CSyn.parsedProgCFGs prog) $ \(C.Reg.AnyCFG defCfg) -> do
-          C.SomeCFG defSsa <- pure $ C.toSSA defCfg
-          -- This could probably be a helper defined in Crucible...
-          let bindCfg c = C.bindFnHandle (C.cfgHandle c) (C.UseCFG c (C.postdomInfo c))
-          bindCfg defSsa
+  let setupHook :: forall sym. Macaw.SetupHook sym arch
+      setupHook = Macaw.syntaxSetupHook la dl cfgs prog
```

Comment:
This definition is now short enough that I wonder if we shouldn't just inline it.

Reply:
I see what you're saying, but the use-site is already a pretty long line so I think I'll keep it. The type signature also highlights that this needs to be polymorphic over `sym`, which is nice.